src/main/java/seedu/address/logic/commands/addstudenttoclasscommands/AddStudentToClassByEmailCommand.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands.addstudenttoclasscommands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.messages.PersonMessages;
+import seedu.address.logic.messages.TutorialClassMessages;
+import seedu.address.model.Model;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.module.ModuleTutorialPair;
+import seedu.address.model.module.TutorialClass;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Person;
+
+/**
+ * Adds a student to a class by email.
+ */
+public class AddStudentToClassByEmailCommand extends AddStudentToClassCommand {
+
+    private final Predicate<Person> predicate;
+
+    private final Email email;
+
+    /**
+     * Adds a student to a class by email.
+     * 
+     * @param email
+     * @param module
+     * @param tutorialClass
+     */
+    public AddStudentToClassByEmailCommand(Email email, ModuleCode module, TutorialClass tutorialClass) {
+        super(module, tutorialClass);
+        this.email = email;
+        this.predicate = person -> person.getEmail().equals(email);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        ModuleTutorialPair moduleAndTutorialClass = getModuleAndTutorialClass(model);
+        TutorialClass tutorialClass = moduleAndTutorialClass.getTutorialClass();
+        ModuleCode module = moduleAndTutorialClass.getModule();
+        Person personToAdd;
+
+        personToAdd = model.searchPersonByPredicate(predicate);
+        if (personToAdd == null) {
+            throw new CommandException(String.format(PersonMessages.MESSAGE_PERSON_EMAIL_NOT_FOUND, email));
+        }
+        System.out.println(tutorialClass.getStudents());
```

Comment:
Is this line neccessary?

Reply:
Yep will take this out. Was meant for debugging